stl/inc/valarray

```diff
@@ -1373,6 +1373,12 @@ public:
         return _Stride;
     }
 
+#if _HAS_CXX20
+    _NODISCARD friend bool operator==(const slice& _Left, const slice& _Right) const noexcept /* strengthened */ {

```

Comment:
This line is causing the tests to fail with:
```
D:\build\out\inc\valarray(1377): error C2270: '==': modifiers not allowed on nonmember functions
```
This is a non-member function taking two `slice`s (unlike a member function taking one `slice` and comparing it to `*this`), so it can't be marked as `const`. That is:

```suggestion
    _NODISCARD friend bool operator==(const slice& _Left, const slice& _Right) noexcept /* strengthened */ {
```

(It's a non-member function even though it's defined within the class definition as a `friend`.)